lib/srv/regular/sshserver.go

```diff
@@ -1174,14 +1174,7 @@ func (s *Server) handleAgentForwardNode(req *ssh.Request, ctx *srv.ServerContext
 		return trace.Wrap(err)
 	}
 
-	// open a channel to the client where the client will serve an agent
-	authChannel, _, err := ctx.Conn.OpenChannel(sshutils.AuthAgentRequest, nil)
-	if err != nil {
-		return trace.Wrap(err)
-	}
-
-	// save the agent in the context so it can be used later
-	ctx.Parent.SetAgent(agent.NewClient(authChannel), authChannel)
+	ctx.Parent.SetForwardAgent(true)
```

Comment:
I wonder if a better name would help clarify what was is happening here? If I understand correctly at this point we want to allocate a lazy agent that will open a channel when the first request comes through. So maybe `ctx.Parent.StartLazyAgent()`?

Reply:
This method doesn't start anything.  It just sets a flag, indicating that the context is now *capable* of agent forwarding (i.e. the client has sent an agent forwarding request and the server has verified that the associated teleport user is allowed to use agent forwarding).  This flag existed implicitly when we used a single global `Agent` because it would be `nil` until the request had gone through.

A boolian flag named `ForwardAgent` seemed like a good way to represent this, but I'm open to alternatives... does `AllowAgentForwarding` get the point across better?